pkg/gds/gds_test.go

```diff
@@ -52,82 +74,136 @@ func loadFixtures(s *gdsTestSuite) {
 			return nil
-		}
-		if strings.HasPrefix(info.Name(), "certreqs::") {
+		case strings.HasPrefix(key, certPrefix):
 			cert := &models.CertificateRequest{}
 			err = protojson.Unmarshal(data, cert)
 			require.NoError(err)
-			dbCerts[info.Name()] = cert
+			s.fixtures[key] = cert
 			return nil
 		}
 		return fmt.Errorf("unrecognized prefix for file: %s", info.Name())
 	})
 	require.NoError(err)
-	os.RemoveAll(root)
-	require.NoError(err)
+	require.NoError(os.RemoveAll(root))
+}
+
+// writeObj writes a protobuf object to the given database.
+func (s *gdsTestSuite) writeObj(db *leveldb.DB, key string, obj interface{}) {
+	var (
+		data []byte
+		err  error
+	)
+	require := s.Require()
+	switch obj := obj.(type) {
+	case *pb.VASP:
+		data, err = proto.Marshal(obj)
+		require.NoError(err)
+	case *models.CertificateRequest:
+		data, err = proto.Marshal(obj)
+		require.NoError(err)
+	default:
+		require.Fail(fmt.Sprintf("unrecognized object for key: %s", key))
+	}
+	require.NoError(db.Put([]byte(key), data, nil))
 }
 
 // generateDB generates an updated database and compresses it to a gzip file.
 // Note: This also generates a temporary directory which the suite teardown
 // should clean up.
-func generateDB(s *gdsTestSuite) {
+func (s *gdsTestSuite) generateDB() {
 	require := s.Require()
-	db, err := leveldb.OpenFile(s.db, nil)
+	db, err := leveldb.OpenFile(s.dbPath, nil)
 	require.NoError(err)
 	defer db.Close()
-
-	loadFixtures(s)
+	small, err := leveldb.OpenFile(s.smallDBPath, nil)
+	require.NoError(err)
+	defer small.Close()
+	s.loadFixtures()
 
 	// Write all the test fixtures to the database.
-	for name, vasp := range dbVASPs {
-		data, err := proto.Marshal(vasp)
-		require.NoError(err)
-		err = db.Put([]byte(name), data, nil)
-		require.NoError(err)
+	for id, obj := range s.fixtures {
+		s.writeObj(db, id, obj)
 	}
+	require.NoError(utils.WriteGzip(s.dbPath, s.dbGzip))
+	log.Info().Str("db", s.dbGzip).Msg("successfully regenerated test database")
 
-	err = utils.WriteGzip(s.db, s.golden)
-	require.NoError(err)
-	log.Info().Str("db", s.golden).Msg("successfully regenerated test database")
+	// Write test fixtures to the small database.
+	for _, id := range smallDBFixtures {
+		obj, ok := s.fixtures[id]
+		require.True(ok)
+		s.writeObj(small, id, obj)
+	}
+	require.NoError(utils.WriteGzip(s.smallDBPath, s.smallDBGzip))
+	log.Info().Str("db", s.smallDBGzip).Msg("successfully regenerated test database")
 }
 
 func (s *gdsTestSuite) SetupSuite() {
 	var err error
 	require := s.Require()
+	gin.SetMode(gin.TestMode)
+
 	s.fakes = filepath.Join("testdata", "fakes.tgz")
-	s.golden = filepath.Join("testdata", "db.tgz")
-	s.db, err = ioutil.TempDir("testdata", "db-*")
+	s.dbGzip = filepath.Join("testdata", "db.tgz")
+	s.smallDBGzip = filepath.Join("testdata", "smalldb.tgz")
+	s.dbPath, err = ioutil.TempDir("testdata", "db-*")
+	require.NoError(err)
+	s.smallDBPath, err = ioutil.TempDir("testdata", "smalldb-*")
 	require.NoError(err)
```

Comment:
Rather than loading all the fixtures at the beginning of the suite, I'm going to create individual cache functions for `loadSmallFixtures()`, `loadAllFixtures()`, and `loadEmptyFixtures()`, then pair those functions with `reset` counterparts in case the test modifies the database. This will prevent the need for the `BeforeTest` action, and allow us to run subsets of tests without having to load unnecessary fixtures. 

Also, I'm going to do a bit of reorganization if that's ok and try to group the fixture loading at the bottom; let me know if this makes sense or if you think we should create a new file for fixtures, e.g. `testdata.go` or `fixtures_test.go` or something like that.

Reply:
I think that organization makes sense for now, if we require more bells and whistles in the fixture loading then we can always move it to another file.